src/domain/synthetics/trade/useTradeboxState.ts

```diff
@@ -598,13 +621,21 @@ export function useTradeboxState(
   );
 
   useEffect(() => {
+    /**
+     * This check required to avoid redundant calls caused by chainId override in context/SynteticsStateContextProvider.txs:104
+     * Overriding chainId causes setStoredOptionsOnChain change and this effect is triggered overriding stored options with wrong chainId
+     */
+    if (isEqual(storedOptions.advanced, advancedOptions)) {
```

Comment:
P.S. I didn't prefer useLatest cuz can't predict possible issues in UI, if reviewer can approve that it would be safe the useLatest also possible solution. But I think that this whole hook (useTradeboxState) should be refactored considering chainId override (or  refuse chainId overriding and solve required things in other maner)